test/integration/index.spec.tsx

```diff
@@ -0,0 +1,27 @@
+import mockServer from './utilities/mockServer'
+import SitePage, { getStaticPaths, getStaticProps } from '../../pages/[[...slug]]/index'
```

Comment:
One other thing is testing these Next.js functions could potentially cause problems if Next.js decides to change how they are implemented / work together. Not necessarily a showstopper right here and now but just to be aware that effectively we are testing and making some assumptions around the Next.js implementations :grin:

Reply:
From my experience this sort of page level integration testing for Next.js doesn't have really have a standard solution. Though I did just have a quick Google and found this package [`next-page-tester`](https://github.com/toomuchdesign/next-page-tester) which seems quite `@testing-library`-esque for this purpose. Seems quite new but we could explore...

Reply:
I tried this `const { render } = await getPage({ route: '/demo/js', })` from the example, but it throw 'catch-all must be the last part of the URL.' It seems super basic and I didn't find any issue with it. Did I miss anything?

Reply:
I tested it on a new project. Same error with catch-all routes `[...slug]` and optional catch-all routes `[[...slug]]`. It only works on static path pages.

Reply:
Bummer that's annoying I guess e2e will still be the best option for dynamic routes.